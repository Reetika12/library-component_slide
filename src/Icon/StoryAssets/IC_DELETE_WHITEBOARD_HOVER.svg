<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none" viewBox="0 0 24 24">
    <path fill="#333" d="M8.032 18.642L7 10h10l-1.032 8.642C15.876 19.417 15.22 20 14.44 20H9.561c-.78 0-1.437-.583-1.53-1.358z"/>
    <rect width="14" height="1.5" x="5" y="8" fill="#333" rx=".75"/>
    <path fill="#333" d="M7 7c0-1.105.895-2 2-2h6c1.105 0 2 .895 2 2v1H7V7z"/>
</svg>
