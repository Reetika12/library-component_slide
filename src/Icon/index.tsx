import React, { CSSProperties, ReactNode, useState } from 'react';

import './icon.scss';

export interface IIconProps {
	normalIcon: ReactNode;
	hoverIcon?: ReactNode;
	activeIcon?: ReactNode;
	activeHoverIcon?: ReactNode;
	isActive?: boolean;
	clickable?: boolean;
	containerStyle?: CSSProperties;
	innerContainerStyle?: CSSProperties;
	innerContainerStyleHover?: CSSProperties;
	innerContainerStyleActive?: CSSProperties;
	innerContainerStyleActiveHover?: CSSProperties;
	onClick?: (e: React.SyntheticEvent) => void;
}

export default function Icon(props: IIconProps): JSX.Element {
	const {
		normalIcon,
		hoverIcon,
		activeIcon,
		isActive,
		activeHoverIcon,
		containerStyle,
		innerContainerStyle,
		innerContainerStyleHover,
		innerContainerStyleActive,
		innerContainerStyleActiveHover,
		clickable,
		onClick,
	} = props;
	const [isHover, setIsHover] = useState(false);

	const hoverFinal = hoverIcon || normalIcon;
	const activeHoverFinal = activeHoverIcon || activeIcon;

	const containerStyleDefault: CSSProperties = {
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'center',
		backgroundColor: 'transparent',
		position: 'relative',
		height: '40px',
		width: '40px',
	};

	if (clickable) {
		containerStyleDefault.cursor = 'pointer';
	}

	const innerContainerStyleDefault: CSSProperties = {};

	const containerStyleFinal = { ...containerStyleDefault, ...containerStyle };
	let innerContainerStyleFinal = { ...innerContainerStyleDefault, ...innerContainerStyle };

	if (isHover) {
		innerContainerStyleFinal = { ...innerContainerStyle, ...innerContainerStyleHover };
	}

	if (isActive) {
		innerContainerStyleFinal = { ...innerContainerStyle, ...innerContainerStyleActive };
	}

	if (isActive && isHover) {
		innerContainerStyleFinal = {
			...innerContainerStyle,
			...innerContainerStyleActiveHover,
		};
	}

	return (
		<div
			style={containerStyleFinal}
			onMouseEnter={() => setIsHover(true)}
			onMouseLeave={() => setIsHover(false)}
			onClick={onClick}
			className="icon"
		>
			<div style={{ top: 0, left: 0, zIndex: 1, position: 'absolute' }}>
				{!isHover && !isActive && normalIcon}
				{isHover && !isActive && hoverFinal}
				{!isHover && isActive && activeIcon}
				{isHover && isActive && activeHoverFinal}
			</div>
			<div style={innerContainerStyleFinal} />
		</div>
	);
}
