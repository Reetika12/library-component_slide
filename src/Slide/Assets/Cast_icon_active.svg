<svg xmlns="http://www.w3.org/2000/svg" width="40" height="40" fill="none" viewBox="0 0 40 40">
    <rect width="24" height="24" fill="#4FC027" rx="12" transform="matrix(-1 0 0 1 32 8)"/>
    <rect width="9" height="9" x="15.5" y="15.5" fill="#fff" rx="2"/>
</svg>
