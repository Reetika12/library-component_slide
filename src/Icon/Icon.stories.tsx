import React from 'react';
import { Story, Meta } from '@storybook/react';

import Icon, { IIconProps } from './index';

import { ReactComponent as PlayIcon } from './StoryAssets/Cast_icon_normal.svg';
import { ReactComponent as PlayIconActive } from './StoryAssets/Cast_icon_active.svg';

import { ReactComponent as ZoomIcon } from './StoryAssets/ZOOMLENCE_ICON.svg';
import { ReactComponent as ZoomIconHover } from './StoryAssets/ZOOMLENCE_ICON_HOVER.svg';

import { ReactComponent as SlideDeteleteIconHover } from './StoryAssets/IC_DELETE_WHITEBOARD_HOVER.svg';

export default {
	title: 'Icon',
	component: Icon,
} as Meta;

const Template: Story<IIconProps> = (args) => <Icon {...args} />;

const commonProps = {
	normalIcon: <PlayIcon />,
	hoverIcon: <ZoomIcon />,
	activeIcon: <PlayIconActive />,
	activeHoverIcon: <SlideDeteleteIconHover />,
	isActive: false,
	clickable: true,
};

export const Basic = Template.bind({});
Basic.args = {
	...commonProps,
};

Basic.storyName = 'Basic';

export const newSlide = Template.bind({});
newSlide.args = {
	normalIcon: <ZoomIcon />,
	hoverIcon: <ZoomIconHover />,
	containerStyle: {
		backgroundColor: '#fff',
		height: '40px',
		width: '40px',
	},
	innerContainerStyle: {
		height: '24px',
		width: '24px',
		backgroundColor: 'rgba(57, 57, 57, .8)',
		borderRadius: '4px',
	},
	innerContainerStyleHover: {
		backgroundColor: 'rgba(0, 0, 0, .8)',
	},
	clickable: true,
};

newSlide.storyName = 'Icon with tappable area';
