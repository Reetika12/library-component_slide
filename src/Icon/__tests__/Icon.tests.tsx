import * as React from 'react';
// import { act } from 'react-dom/test-utils';
// import * as ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Icon from '../index';

import { ReactComponent as PlayIcon } from '../StoryAssets/Cast_icon_normal.svg';
import { ReactComponent as PlayIconActive } from '../StoryAssets/Cast_icon_active.svg';

import { ReactComponent as ZoomIcon } from '../StoryAssets/ZOOMLENCE_ICON.svg';

import { ReactComponent as SlideDeteleteIconHover } from '../StoryAssets/IC_DELETE_WHITEBOARD_HOVER.svg';

describe('Icon', function test() {
	it('should render correctly', () => {
		const component = shallow(
			<Icon
				normalIcon={<PlayIcon />}
				hoverIcon={<ZoomIcon />}
				activeIcon={<PlayIconActive />}
				activeHoverIcon={<SlideDeteleteIconHover />}
				isActive={false}
				clickable
			/>
		);

		expect(component).toMatchSnapshot();
	});

	it('should render correctly for tappable area', () => {
		const component = shallow(
			<Icon
				normalIcon={<PlayIcon />}
				hoverIcon={<ZoomIcon />}
				activeIcon={<PlayIconActive />}
				activeHoverIcon={<SlideDeteleteIconHover />}
				isActive={false}
				clickable
				containerStyle={{
					height: '100px',
					width: '100px',
				}}
				innerContainerStyle={{
					height: '24px',
					width: '24px',
				}}
			/>
		);

		expect(component).toMatchSnapshot();
	});
});
