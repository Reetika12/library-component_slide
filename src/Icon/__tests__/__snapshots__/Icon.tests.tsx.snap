// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render correctly 1`] = `
<div
  className="icon"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "cursor": "pointer",
      "display": "flex",
      "height": "40px",
      "justifyContent": "center",
      "position": "relative",
      "width": "40px",
    }
  }
>
  <div
    style={
      Object {
        "left": 0,
        "position": "absolute",
        "top": 0,
        "zIndex": 1,
      }
    }
  >
    <div />
  </div>
  <div
    style={Object {}}
  />
</div>
`;

exports[`Icon should render correctly for tappable area 1`] = `
<div
  className="icon"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "cursor": "pointer",
      "display": "flex",
      "height": "100px",
      "justifyContent": "center",
      "position": "relative",
      "width": "100px",
    }
  }
>
  <div
    style={
      Object {
        "left": 0,
        "position": "absolute",
        "top": 0,
        "zIndex": 1,
      }
    }
  >
    <div />
  </div>
  <div
    style={
      Object {
        "height": "24px",
        "width": "24px",
      }
    }
  />
</div>
`;
